tests/queries/0_stateless/02815_first_line.sql

```diff
@@ -0,0 +1,12 @@
+select firstLine('foo\nbar\nbaz');
+select firstLine('foo\rbar\rbaz');
+select firstLine('foo\r\nbar\r\nbaz');
+select firstLine('foobarbaz');
+
+select '== vector';
+
+drop table if exists 02815_first_line_vector;
+create table 02815_first_line_vector (n Int32, text String) engine = MergeTree order by n;
+
+insert into 02815_first_line_vector values (1, 'foo\nbar\nbaz'), (2, 'quux\n'), (3, 'single line'), (4, 'windows\r\nline breaks');
```

Comment:
For one line you can just `select firstLine(materialize('foo\nbar\nbaz'))`

Reply:
Thanks, that's good to know!
I'll probably prefer leaving a table with multiple values because it creates a tiny chance it will trigger the `vector` handler for the function. Do you know btw if it can be triggered deterministically?

Reply:
If this case (from current test) with three lines (ColumnString) is meant, then I don’t know. Or what vector do you mean?
(additionally FunctionStringToString has useDefaultImplementationForConstants = true, which means that the Const and non Const expression will be treated the same way.)